### PR TITLE
rpcbinding: export any unexported fields of structures/events

### LINKED ISSUE #3083: Check and convert private fields of RPC binding structures to public
Resulting exported RPC binding structures must not contain private fields. This issue includes two tasks:
1. Check exported structures of the source contract code and convert all private names to exported ones.
2. If generating RPC binding not from the Go code, then ensure that all private fields declared in the contract manifest must successfully be converted to the public ones.

Ref. https://github.com/nspcc-dev/neofs-contract/blob/84ff5d244f6913987c2fb7dc3e2adfb5dfa81381/rpc/container/rpcbinding.go#L27.

### PR DESCRIPTION
**Pull request by AnnaShaleva**
Perform private -> public transformation at the last step of RPC binding generation so that it works not only with NeoGo contracts, but with any other contracts.

Close #3083.

Note: only the second part of #3083 is done, as suggested in https://github.com/nspcc-dev/neo-go/issues/3083#issuecomment-1667617080. However, this code will generate garbadge output for Go contracts where the following structures are present:
```
type SomeStr struct {
    I int
    i []byte
}
```
If we'd like to avoid this, I'd suggest implementing some compile-time warning about such structures/events. Another possible solution is to check if these situations are present at a binding generation time and return warning if so. @roman-khimov, what do you think, do we need it?

### CODE REVIEW

File: cli/smartcontract/testdata/notifications/rpcbindings_guessed.out [hunk 1]
```diff
@@ -103,6 +103,11 @@ type SomeArrayEvent struct {
 	A [][]*big.Int
 }
 
+// SomeUnexportedFieldEvent represents "SomeUnexportedField" event emitted by the contract.
+type SomeUnexportedFieldEvent struct {
+	S *Unnamed
```
Comment (AnnaShaleva) [inline cli/smartcontract/testdata/notifications/rpcbindings_guessed.out:108]: We already have another `Unnamed` structure in this package nested from other event. This bug will be fixed in a separate issue.

Comment (AnnaShaleva) [reply cli/smartcontract/testdata/notifications/rpcbindings_guessed.out:108]: Ref. https://github.com/nspcc-dev/neo-go/issues/3088. Let it be "as is" in this PR.

**Comment by roman-khimov** [comment]
> check if these situations are present at a binding generation time and return warning if so

Config can have anything and you have to deal with it.

File: pkg/smartcontract/rpcbinding/binding.go [hunk 2]
```diff
@@ -863,7 +864,7 @@ func toTypeName(s string) string {
 			return -1
 		}
 		return c
-	}, strings.ToUpper(s[0:1])+s[1:])
+	}, upperFirst(s))
```
Comment (roman-khimov) [inline pkg/smartcontract/rpcbinding/binding.go:867]: Unrelated?

Comment (AnnaShaleva) [reply pkg/smartcontract/rpcbinding/binding.go:889]: Yes, moved to a separate commit.

**Review by roman-khimov — APPROVED** [review]

**Comment by roman-khimov** [comment]
Please fix the linter.
